extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -287,6 +287,12 @@ export class AmpSlideScroll extends BaseSlides {
   }
 
   /** @override */
+  unlayoutCallback() {
+    this.slideIndex_ = null;
```

Comment:
@camelburrito Yes, it does solve the issue. I'll add a unit test as per your request.